src/executables/util_sa.c

```diff
@@ -1739,11 +1740,16 @@ diagdb (UTIL_FUNCTION_ARG * arg)
 		  goto error_exit;
 		}
 	    }
-	  if (heap_dump_heap_file (thread_p, outfp, dump_records, class_name) == ER_LC_UNKNOWN_CLASSNAME)
+	  error_code = heap_dump_heap_file (thread_p, outfp, dump_records, class_name);
+	  if (error_code != NO_ERROR)
 	    {
-	      PRINT_AND_LOG_ERR_MSG (msgcat_message
-				     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_DIAGDB, DIAGDB_MSG_UNKNOWN_CLASS),
-				     class_name);
+	      if (error_code == ER_LC_UNKNOWN_CLASSNAME)
+		{
+		  PRINT_AND_LOG_ERR_MSG (msgcat_message
+					 (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_DIAGDB, DIAGDB_MSG_UNKNOWN_CLASS),
+					 class_name);
+		}
+	      goto error_exit;
```

Comment:
Ah, one thing I missed here is that we should call `db_shutdown ()` ;; 
We can call `db_shutdown ()` by removing this `goto` statement or just call the `db_shutdown ()` above the `goto` label

Reply:
```suggestion     
	      db_shutdown ();
	      goto error_exit;  
```

I think `goto` statement in above (line 1740) also needs `db_shutdown ()`.